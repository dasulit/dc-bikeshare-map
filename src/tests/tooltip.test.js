import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import Tooltip from '../components/tooltip';

describe('Tooltip', () => {
  test('renders null if no features provided', () => {
    const wrapper = shallow(
      React.createElement(Tooltip, {
        featureProperties: {}
      })
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('renders with Capital Bikeshare properties', () => {
    const wrapper = shallow(
      React.createElement(Tooltip, {
        featureProperties: {
          provider: 'capital-bikeshare',
          name: 'Florida Ave & R St NW',
          bikes: 12,
          docks: 3
        }
      })
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('renders with JUMP properties', () => {
    const wrapper = shallow(
      React.createElement(Tooltip, {
        featureProperties: {
          provider: 'jump',
          battery: '100%'
        }
      })
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
