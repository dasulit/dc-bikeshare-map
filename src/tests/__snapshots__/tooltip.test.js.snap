// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders null if no features provided 1`] = `""`;

exports[`Tooltip renders with Capital Bikeshare properties 1`] = `
<div
  className="tooltip capital-bikeshare"
>
  <b>
    Capital Bikeshare station
  </b>
  <div
    className="margin-bottom-10"
  >
    Florida Ave & R St NW
  </div>
  <ul>
    <li>
      Available bikes: 
      12
    </li>
    <li>
      Available docks: 
      3
    </li>
  </ul>
</div>
`;

exports[`Tooltip renders with JUMP properties 1`] = `
<div
  className="tooltip jump"
>
  <b>
    JUMP Bike
  </b>
   
  100%
   battery
</div>
`;
